dataset/src/test/scala/frameless/SortTests.scala

```diff
@@ -0,0 +1,141 @@
+package frameless
+
+import org.apache.spark.sql.FramelessInternals.UserDefinedType
+import org.apache.spark.sql.catalyst.util.{ ArrayBasedMapData, DateTimeUtils }
+import org.apache.spark.sql.types._
+import org.apache.spark.sql.{ functions => sfunc }
+import org.scalacheck.Prop
+import org.scalacheck.Prop._
+import shapeless.test.illTyped
+
+@SQLUserDefinedType(udt = classOf[UdtMapEncoded])
+class UdtMapClass(val a: Int) {
+  override def equals(other: Any): Boolean = other match {
+    case that: UdtMapClass => a == that.a
+    case _ => false
+  }
+
+  override def hashCode(): Int = {
+    val state = Seq[Any](a)
+    state.map(_.hashCode()).foldLeft(0)((a, b) => 31 * a + b)
+  }
+
+  override def toString = s"UdtMapClass($a)"
+}
+
+object UdtMapClass {
+  implicit val urtEncoderClass = new UdtMapEncoded
+}
+
+class UdtMapEncoded extends UserDefinedType[UdtMapClass] {
+  override def sqlType: DataType = MapType(StringType, IntegerType)
+  override def serialize(obj: UdtMapClass): Any =
+    ArrayBasedMapData(Map.empty)
+  override def deserialize(datum: Any): UdtMapClass =
+    new UdtMapClass(1)
+  override def userClass: Class[UdtMapClass] = classOf[UdtMapClass]
+}
+
+class SortTests extends TypedDatasetSuite {
+  test("bad udt") {
```

Comment:
I could use some help with this. Unclear how to support `UDT`s via `Record` types etc while preventing those whose underlying `sqlType` are not sortable. The generic `Record` support is super helpful for `StructType`s. 
Any ideas?